### ISSUE TITLE
There is no document ？

### ISSUE THREAD

**Issue by frodoking** [gin-gonic/gin#366]
hi guys,

Gin is great Go Project, special use tree to manage router,
like use ':' & '*' to match params. but when i read tree.go. 
i can't understand how to build a MethodTree, because there are so many complex rules.
Which would have been fine if the document have been built on your website(https://gin-gonic.github.io).
Additionnal, perfer to see some Structure Chart or Timing Chart about this project.
I want to see it in the back. 

thanks.

**Comment by javierprovecho** [comment]
@frodoking can you rephrase your question, please?

**Comment by frodoking** [comment]
Please forgive me for my poor English.

gin is a great project, it is special when using tree structure to manage the path, the symbol of ':' and '*' to match the param of queries. But when I go through the source code of tree.go, I can not understand how to build methodTree, because the rule of tree architecture is too complex. So if it is better to offer some documents in your website, and I also hope to look up project related pictures for the whole architecture and timing order.

thanks

**Comment by javierprovecho** [comment]
@frodoking thank you!

For benchmark times, you can checkout a wonderful table at `README.md`.

For more information about tree structure used in the router, I suggest to checkout the original project at https://github.com/julienschmidt/httprouter
